pkg/nfd-master/metrics.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nfdmaster
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/node-feature-discovery/pkg/version"
+)
+
+// When adding metric names, see https://prometheus.io/docs/practices/naming/#metric-names
+const (
+	buildInfoQuery         = "nfd_master_build_info"
+	updatedNodesQuery      = "nfd_updated_nodes"
+	crdProcessingTimeQuery = "nfd_crd_processing_time"
```

Comment:
I think we'd really want some error counters as well (https://github.com/kubernetes-sigs/node-feature-discovery/pull/1242#pullrequestreview-1471971539). Those would be super valuable for observability, detecting problems. At least node update failures and NodeFeatureRule processing errors. We could also add counters for invalid labels, ERs and taints (that got filtered out) before updating the node

Reply:
Could these "extra" metrics be a follow up PR, and this one to focus on basic,raw, functionality?  I am more than happy to create an issue and assign it to my self as a follow up

Reply:
I think that's fine